Applications/Utils/FileConverter/GocadTSurfaceReader.cpp

```diff
@@ -51,21 +51,46 @@ int main(int argc, char* argv[])
         "if set, OGS-Meshes will be written in binary format");
     cmd.add(write_binary_arg);
 
+    TCLAP::SwitchArg export_lines_arg(
+        "l", "lines-only",
+        "if set, only PLine datasets will be parsed from the input file");
+    cmd.add(export_lines_arg);
+
+    TCLAP::SwitchArg export_surfaces_arg(
+        "s", "surfaces-only",
+        "if set, only TSurf datasets will be parsed from the input file");
+    cmd.add(export_surfaces_arg);
+
     cmd.parse(argc, argv);
 
+    if (export_lines_arg.isSet() && export_surfaces_arg.isSet())
```

Comment:
Would it be possible to use tclap's functionality here? http://tclap.sourceforge.net/manual.html#XOR

Reply:
I've added 

```
cmd.xorAdd(export_lines_arg, export_surfaces_arg);
```

but this causes the application to crash?!

Also, I am not sure if XOR means that one of these arguments *has* to set. For my app that isn't the case.